### PR TITLE
NumberInput arrow click issue

### PR DESCRIPTION
**Pull request by ronronscelestes**
## What

Fixed NumberInput not activating onValueChange when clicking on arrows up/down
Added disabled state 
Added tests

## Demo

Click on arrows to change the value
The value should update immediately (not on blur)

### CODE REVIEW

File: packages/strapi-design-system/src/NumberInput/NumberInput.js [hunk 1]
```diff
@@ -124,10 +146,29 @@ export const NumberInput = React.forwardRef(
...
+                <ArrowButton
+                  aria-hidden
+                  reverse
+                  onClick={() => {
+                    // increment needs an argument, so we can't remove the parenthesis
+                    increment();
+                  }}
+                  tabIndex={-1}
+                  type="button"
+                  data-testid="ArrowUp"
```
Comment (HichamELBSI) [inline packages/strapi-design-system/src/NumberInput/NumberInput.js:158]: Don't we need to add the disabled props here? To fix the disabled state

File: packages/strapi-design-system/src/NumberInput/NumberInput.js [hunk 2]
```diff
@@ -124,10 +146,29 @@ export const NumberInput = React.forwardRef(
...
-                <ArrowButton aria-hidden onClick={decrement} tabIndex={-1} type="button">
+                <ArrowButton
+                  aria-hidden
+                  onClick={() => {
+                    // decrement needs an argument, so we can't remove the parenthesis
+                    decrement();
+                  }}
+                  tabIndex={-1}
+                  type="button"
+                  data-testid="ArrowDown"
```
Comment (HichamELBSI) [inline packages/strapi-design-system/src/NumberInput/NumberInput.js:170]: same question

File: packages/strapi-design-system/src/NumberInput/NumberInput.js (hunk 1 above)
Comment (ronronscelestes) [reply packages/strapi-design-system/src/NumberInput/NumberInput.js:158]: you're right good catch, on it!

**Review by mfrachet — APPROVED** [review]
go for it when tests pass 👍🏻

**Pull request merged**